src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -381,12 +381,18 @@ public void setName(String name) {
      * @return interned string
      */
     protected String getCachedString(String str) {
-        WeakReference<String> internedString = stringCache.get(str);
-        if ( internedString == null || internedString.get() == null) {
-            internedString = new WeakReference<>(str);
-            stringCache.put(str, internedString);
+        if (!vcfIndexUseStringsCache)
+            return str;
+        WeakReference<String> stringWeakReference = stringCache.get(str);
```

Comment:
Maybe I'm confused, but I was thinking of running SortVcf. Looking at the code, it seems that most of it has to do with collecting multiple inputs...so i retract my statement. it can easily be done within htsjdk.